core/templates/filters/translate.pipe.ts

```diff
@@ -0,0 +1,152 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Translate pipe for i18n translations.
+ */
+
+import { ChangeDetectorRef, OnDestroy, Pipe, PipeTransform }
+  from '@angular/core';
+import { Subscription } from 'rxjs';
+import { LangChangeEvent, TranslateService } from 'services/translate.service';
+import { UtilsService } from 'services/utils.service';
+
+/**
+ * Commonly used terms in this file:
```

Comment:
I just want to say that this is an order of magnitude better than the original. Thanks @srijanreddy98!

Reply:
This should have been the quality from the start. I try to at least match this quality before requesting a review for my future prs.